### PR TITLE
Move stage cred to cephci template file

### PR DESCRIPTION
**Pull request by VasishtaShastry**
Move stage cred to cephci template file

Ref - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1612617757424/install_ceph_pre-requisites_0.log

Signed-off-by: VasishtaShastry <vipin.indiasmg@gmail.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Review by psathyan — DISMISSED** [review]

**Comment by sunilangadi2** [comment]
@VasishtaShastry where can we set this value `is_live` to True ?
Is this works fine with anisble suite along with dashboard?

**Comment by VasishtaShastry** [comment]
@sunilangadi2 
> @VasishtaShastry where can we set this value `is_live` to True ?
It needs to be set in suite file under , something like
```
      config:
        is_live: true

```
> Is this works fine with ansible suite along with dashboard?
I think it should as I have not made changes in other spaces than with subscribing to stage

File: cephci.yaml.template
Comment (ktdreyer) [inline cephci.yaml.template:23]:
```suggestion
# access to rhel and ceph repos that are configured on the staging CDN.
```

File: tests/misc_env/install_prereq.py
Comment (ktdreyer) [inline tests/misc_env/install_prereq.py:45]: "live" is abiguous to me. Can we rename this setting to `is_production` instead? Is this "live" name used anywhere else (in any internal documentation, etc)?

File: cephci.yaml.template
Comment (ktdreyer) [reply cephci.yaml.template:23]: As a new user, how would I determine whether I need to configure this option or not? In other words, what should I do in order to determine if my suite needs this?

Comment (VasishtaShastry) [reply cephci.yaml.template:23]: `All suites use staging CDN unless is_live config (or is_production) is set to true` 
will add it in template file

File: tests/misc_env/install_prereq.py
Comment (VasishtaShastry) [reply tests/misc_env/install_prereq.py:45]: I agree that 'live' can be replaced with 'production' to make it unambiguous.

**Review by psathyan — APPROVED** [review]

File: tests/misc_env/install_prereq.py
Comment (ktdreyer) [inline tests/misc_env/install_prereq.py:141]: I don't think we should use `.get()` unless it is truly an optional setting.

If a user lacks `cdn_credentials` or `stage_credentials` in `~/.cephci.yaml`, and we call into `setup_subscription_manager()`, what is the expected behavior?

File: tests/misc_env/install_prereq.py
Comment (psathyan) [inline tests/misc_env/install_prereq.py:147]: IMO, we could retrieve the contents from the yaml file and then use the appropriate key for extracting the values. For example:
```
config_ = get_cephci_config()
cmd = "subscription-manager --force register"

if is_production:
  username_ = config_["cdn_credentials"]["username"]
  password_ = config_["cdn_credentials"["password"]
else:
  cmd += " --serverurl=subscription.rhsm.stage.redhat.com:443/subscription --baseurl=https://cdn.redhat.com"
  username_ = config_["stage_credentials"]["username"]
  password_ = config_["stage_credentials"]["password"]

cmd += f" --username={username_} --password={password_}"

ceph.exec_command(sudo=True, cmd=cmd, timeout=720)
```

File: tests/misc_env/install_prereq.py
Comment (psathyan) [inline tests/misc_env/install_prereq.py:173]: In conjunction with the above
```
except AttributeError:
  log.error("Require the %s credential", "cdn_credentials" if is_production else "stage_credentails")
```

**Comment by veera-raghava-reddy** [comment]
Any update on this PR?

**Comment by psathyan** [comment]
@VasishtaShastry could we close on this PR? It's been pending for sometime, let's discuss if you need any further clarification.

**Comment by psathyan** [comment]
@VasishtaShastry lets work together to close on this PR. This is holding up another dependent PR...

**Comment by psathyan** [comment]
@VasishtaShastry lets work together to close on this PR

File: tests/misc_env/install_prereq.py
Comment (VasishtaShastry) [reply tests/misc_env/install_prereq.py:179]: Thank you for the suggestions, I've made changes as suggested

File: tests/misc_env/install_prereq.py
Comment (VasishtaShastry) [reply tests/misc_env/install_prereq.py]: @ktdreyer .get has been avoided
If user miss credentials, newly added Attribute Error would catch and throw error

File: tests/misc_env/install_prereq.py
Comment (VasishtaShastry) [reply tests/misc_env/install_prereq.py]: Fixed the mishap, Marking as resolved
Please reopen if needed

File: utility/utils.py
Comment (VasishtaShastry) [reply utility/utils.py]: This has been implemented
Please review

File: tests/misc_env/install_prereq.py
Comment (VasishtaShastry) [reply tests/misc_env/install_prereq.py]: Thanks a lot for the suggestions, I've made changes as suggested

**Comment by VasishtaShastry** [comment]
@veera-raghava-reddy @psathyan 

Sorry for the delay, Got stuck with Upgrade automation and other tasks and got delayed when I faced env issues in between.

Logs -
Stage - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1622040405201/install_ceph_pre-requisites_0.log
Live - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1622041330688/install_ceph_pre-requisites_0.log

Live has failed but for the scope of this PR I think it looks sufficient. Issue with live seems more of a resource issue and also linked with mail conversation with subject `Need Help : access to CDN live testing with dedicated profile`

1) I've not done negative testing of the code in this PR, will update shortly on that
2) Also will update conf file and suite created for this shortly
3) I'll prepare small writeup on how this PR in master would change usage for *all users* to be updated to confluence page and to team

**Comment by VasishtaShastry** [comment]
Tried negative scenario (User missing credentials) and made some changes in https://github.com/red-hat-storage/cephci/pull/339/files#diff-a2a99cf39f88141c6551d674cc418fa009455f8f3717b41f5a126e80f1097515R174-R180

If CDN credentials missed - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1622445239138
Is stage credentials missed - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1622446013783

Conf and suites used - http://magna002.ceph.redhat.com/ceph-qe-logs/vasishta/PR_339/

**Comment by VasishtaShastry** [comment]
@psathyan I'll update rhcsqe/CephCI+Start+Guide (confluence and team (existing users) once the PR is ready to be merged

**Review by psathyan — APPROVED** [review]

**Pull request merged**
